package messages;


message PVisionField
{

   optional int64 timestamp = 1;
   optional PVisualGoalPost goal_post_l = 2;
   optional PVisualGoalPost goal_post_r = 3;
   repeated PVisualCorner visual_corner = 4;
   repeated PVisualLine visual_line = 5;
   optional PVisualDetection visual_cross = 6;
   
      message Point
      {
	optional float x = 1;
	optional float y = 2;
      }

      message PVisualDetection
      {
	optional float distance = 1;
	optional float bearing = 2;
	repeated Point concrete_coords = 3;
      }
      
   message PVisualGoalPost
   {
      optional PVisualDetection visual_detection = 1;
      optional int32 height = 2;
      optional int32 width = 3;
   }

   message PVisualCorner
   {
      optional PVisualDetection visual_detection = 1;
      optional float orientation = 2;
      optional int32 corner_type = 3;
   }

   message PVisualLine
   {
      optional PVisualDetection visual_detection = 1;
      optional int32 line_id = 2;
      optional int32 start_x = 3;
      optional int32 start_y = 4;
      optional int32 end_x = 5;
      optional int32 end_y = 6;
   }



}